relay/channel/xai/adaptor_test.go

```
@@ -0,0 +1,170 @@
+	request := converted.(ImageRequest)
+	require.Len(t, request.Images, 2)
+	require.Nil(t, request.Image)
+	require.True(t, strings.HasSuffix(request.Images[0].URL, base64.StdEncoding.EncodeToString([]byte("one"))))
+	require.True(t, strings.HasSuffix(request.Images[1].URL, base64.StdEncoding.EncodeToString([]byte("two"))))
+}
+
+func TestConvertImageRequestRejectsUnsupportedMaskAndMissingImage(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	newContext := func() *gin.Context {
+		c, _ := gin.CreateTestContext(httptest.NewRecorder())
+		c.Request = httptest.NewRequest(http.MethodPost, "/v1/images/edits", bytes.NewBufferString(`{"model":"grok-imagine-image","prompt":"edit"}`))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)